### DISCUSSION TITLE
determine type of variable (register)

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#161] [Q&A]
For the c2x, c2d,c2b functions, I need to know which type the input variable has. Do we have a type function yet? I hoped GETTP will do the job, but it always returns 0. If we don't have one, I'll set up a new instruction.

**Comment by adesutherland** [comment]
At the moment I don't 100% know how this is going to work for classic REXX. GETTP will be part of the answer but remember that the compiler has to set it (which it doesn't yet!). We need the class and function overloading mechanism to be defined ...

In the meantime, you will have to define different function [names] for different type - and we will have to fix later - sorry :-(

**Reply by Peter-Jacob** [reply]
No problem. To state my question more precisely, if we have 2 load statements: 
load  r1,1
load r2,"this is a string 
The load sets it exactly with set_const_string() or set_int(), but there is no flag set: this is a string/integer, at least I couldn't find one. 
In my specific case I could say if it's an integer,  convert it to string (ITOS) or it's already a string.

**Reply by adesutherland** [reply]
Understood - gettp/settp is designed to be set by the compiler / assembler; see https://github.com/adesutherland/CREXX/wiki/REXX-Assembler-(RXAS)#dynamic-type-instructions.

What we need is a way for the right function to be called depending on operator type - polymorphism.

The keyhole optimiser is done and I am just doing some jump optimisation - then I can work on how this will be handled next week. I will be on holiday (and being a geek), the cREXX class design will have the code name, Davaar Island

**Comment by Peter-Jacob** [comment]
No urgency, I just wanted to understand itVon meinem Huawei-Telefon gesendet-------- Ursprüngliche Nachricht --------Von: Adrian Sutherland ***@***.***>Datum: Mi., 20. Okt. 2021, 19:31An: adesutherland/CREXX ***@***.***>Cc: Peter-Jacob ***@***.***>, Author ***@***.***>Betreff: Re: [adesutherland/CREXX] determine type of variable (register) (Discussion #161)
Understood - gettp/settp is designed to be set by the compiler / assembler; see https://github.com/adesutherland/CREXX/wiki/REXX-Assembler-(RXAS)#dynamic-type-instructions.
What we need is a way for the right function to be called depending on operator type - polymorphism.
The keyhole optimiser is done and I am just doing some jump optimisation - then I can work on how this will be handled next week. I will be on holiday (and being a geek), the cREXX class design will have the code name, Davaar Island
